holoviews/plotting/mpl/raster.py

```diff
@@ -156,18 +153,25 @@ def init_artists(self, ax, plot_args, plot_kwargs):
 
     def get_data(self, element, ranges, style):
         _, style, axis_kwargs = super(HeatMapPlot, self).get_data(element, ranges, style)
-        mask = np.logical_not(np.isfinite(element.raster))
-        data = np.ma.array(element.raster, mask=mask)
-        style['annotations'] = self._annotate_values(element)
+        aggregate = element.gridded
+        data = np.flipud(aggregate.dimension_values(2, flat=False))
+        shape = data.shape
+        cmap_name = style.pop('cmap', None)
+        cmap = copy.copy(plt.cm.get_cmap('gray' if cmap_name is None else cmap_name))
+        cmap.set_bad('w', 1.)
```

Comment:
Again good point, indeed we already expose this via ``clipping_colors``, should hook that in here.